src/cmap/connect.ts

```diff
@@ -148,7 +149,7 @@ function performInitialHandshake(
 
       if ('isWritablePrimary' in response) {
         // Provide pre-hello-style response document.
-        response.ismaster = response.isWritablePrimary;
+        response[LEGACY_HELLO_COMMAND] = response.isWritablePrimary;
```

Comment:
I think this is what @nbbeeken noted when we were refining this ticket. I have this note: "Current code ismaster.ismaster = ismaster.isWritablePrimary. Need to flip this."

Can someone provide more context on why we want to flip this?

Reply:
Whether the driver is connected to an old server or new server, we want to be able to write code that works uniformly across new / old servers. So instead of having this code that patches on a legacy key when communicating with a server that uses the new isWritablePrimary key, we can patch always patch in the new key if its not present. 

example:
```ts
if (!('isWritablePrimary' in response)) {
  // Provide hello-style response document.
  response.isWritablePrimary = response[LEGACY_HELLO_COMMAND];
}
```
With the above, if we're talking to a new server, isWritablePrimary is defined, so this doesn't run, but on an old server is we will attach it. Now the rest of the driver can always assume that the new key is present.

Reply:
Gotcha. Thanks! 

Does this have a cascading effect?  Meaning, are their places in the driver code where it's looking for ismaster and not isWritablePrimary? The tests are passing on my machine, so I'm thinking this is probably OK...just wanted to verify with you.

Reply:
It's actually not used much, look for where you've changed to `hello[LEGACY_HELLO_COMMAND]`, I think I see two instances. Like the one in server description where we check for both, we could now just check for the new key.

Reply:
Thanks for the info! I made the changes, and the tests are failing (without even tryin the delete response you recommended). I've started debugging but haven't figured it out yet. I'll do more debugging on Monday.

Reply:
I started with fresh eyes today. Here is what I've learned.
- The check in `server_description` is checking if the server is a primary (`ismaster` or `isWritablePrimary`), so this one can be changed. In order to get the tests to pass with this change, I had to update line 171 of `test/functional/unit-sdam/server_selection/spec.test.js` to set `isWritablePrimary` instead of `ismaster`. I *think* this is OK, since this is a fake hello, and I'm not manipulating something coming back from a real server. Can you double check?
- I think the other one you were referring to is `isHello()` in `utils.ts`. `isHello()` is checking if the request is a hello command -- it's checking for `hello` or `ismaster` NOT `isWritablePrimary` or `ismaster`. I think we need to leave this one alone. Let me know if there is another file that I'm overlooking.

I added `delete response[LEGACY_HELLO_COMMAND];` in the two places you suggested:
- Patch with 3.6-replica_set: https://spruce.mongodb.com/version/61dc56b6c9ec4475aa752c58/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC. The tests are passing
- Patch with 5.0-sharded cluster: https://spruce.mongodb.com/version/61dc5ec161837d18207b9984/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC. The tests are passing but timing out with Context: Canceled.

Reply:
We'll have to take a closer look at the hangs, 5.0 is the server version that speaks in hello so I wonder why it would behave differently, but everything else lgtm, one comment below tho, maybe it relates to the hangs.